apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-description/edit.js

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { PlainText } from '@wordpress/editor';
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+
+class SiteDescriptionEdit extends Component {
+	state = {
+		fromApi: null,
+		value: __( 'Site description loading…' ),
+	};
+
+	componentDidMount() {
+		try {
+			apiFetch( { path: '/wp/v2/settings' } ).then( ( { description } ) => {
+				this.setState( {
+					fromApi: description,
+					value: description,
+				} );
+			} );
+		} catch ( error ) {
+			this.handleApiError();
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		if (
+			prevProps.isSaving === false &&
+			this.props.isSaving === true &&
+			this.state.fromApi !== this.state.value
+		) {
+			try {
+				apiFetch( {
+					path: '/wp/v2/settings',
+					method: 'POST',
+					data: {
+						description: this.state.value,
+					},
+				} );
+			} catch ( error ) {
+				this.handleApiError( true );
+			}
+		}
+	}
+
+	handleApiError( post = false ) {
+		if ( post ) {
+			this.setState( {
+				value: this.state.fromApi,
+			} );
+		}
+		// trigger notice
+	}
+
+	render() {
+		const { isSelected } = this.props;
+		if ( isSelected ) {
+			return (
+				<PlainText
+					className="wp-block-a8c-site-description"
+					value={ this.state.value }
+					onChange={ value => this.setState( { value } ) }
```

Comment:
We should also update an attribute of this block when we change the site description within the block. Otherwise, the editor won't be aware that the page content has changed and the "Update" button will remain disabled:

![Jun-13-2019 14-18-15](https://user-images.githubusercontent.com/1233880/59405646-27215080-8de6-11e9-8db8-0b52736803c9.gif)

Reply:
Oh, interesting; my Update button is always available (although that makes no sense either 🙃 ). Will dig further. But we don't want to be saving attributes for this block (we shouldn't ever see `<!-- wp:a8c/site-description {"description":"tagline"} /-->` in markup for instance) 🤔

Reply:
https://github.com/Automattic/wp-calypso/pull/33761#discussion_r293628695